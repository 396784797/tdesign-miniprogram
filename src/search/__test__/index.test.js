import simulate from 'miniprogram-simulate';
import path from 'path';

describe('search', () => {
  const search = simulate.load(path.resolve(__dirname, `../search`), 't-search', {
    less: true,
    rootPath: path.resolve(__dirname, '../..'),
  });

  it(':base', () => {
    const id = simulate.load({
      template: `<t-search></t-search>`,
      usingComponents: {
        't-search': search,
      },
    });
    const comp = simulate.render(id);
    comp.attach(document.createElement('parent-wrapper'));

    expect(comp.toJSON()).toMatchSnapshot();
  });
});
