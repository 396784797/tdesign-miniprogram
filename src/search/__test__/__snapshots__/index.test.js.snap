// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search :base 1`] = `
<main>
  <t-search>
    <wx-view
      class="t-search t-class"
    >
      <wx-view
        class="t-search__input-box t-not-focused t-search__input-box-- t-search__input-box--square t-class-input-container"
      >
        <t-icon
          class="t-icon t-class-left"
        >
          <wx-label
            class="t-class t-icon t-icon-search t-icon-base"
            style=";font-size:20px;"
            bind:tap="onTap"
          />
        </t-icon>
        <wx-input
          class="t-input__keyword t-class-input"
          confirmType="search"
          disabled="{{false}}"
          focus="{{false}}"
          name="input"
          placeholder=""
          placeholderClass="t-search__placeholder t-search__placeholder--normal"
          type="text"
          value=""
          bind:blur="onBlur"
          bind:confirm="onConfirm"
          bind:focus="onFocus"
          bind:input="onInput"
        />
      </wx-view>
    </wx-view>
  </t-search>
</main>
`;
